lib/status-bar/index.js

```diff
@@ -0,0 +1,83 @@
+/* @flow */
+
+import { CompositeDisposable } from 'sb-event-kit'
+import disposify from 'disposify'
+
+import Element from './element'
+import { $file } from '../helpers'
+import type { LinterMessage } from '../types'
+
+export default class StatusBar {
+  element: Element;
+  messages: Array<LinterMessage>;
+  subscriptions: CompositeDisposable;
+  statusBarRepresents: 'Entire Project' | 'Current File';
+  statusBarClickBehavior: 'Toggle Panel' | 'Jump to next error';
+
+  constructor() {
+    this.element = new Element()
+    this.messages = []
+    this.subscriptions = new CompositeDisposable()
+
+    this.subscriptions.add(this.element)
+    this.subscriptions.add(atom.config.observe('linter-ui-default.statusBarRepresents', statusBarRepresents => {
+      const shouldUpdate = typeof this.statusBarRepresents !== 'undefined'
+      this.statusBarRepresents = statusBarRepresents
+      if (shouldUpdate) {
+        this.update(this.messages)
+      }
+    }))
+    this.subscriptions.add(atom.config.observe('linter-ui-default.statusBarClickBehavior', statusBarClickBehavior => {
+      const shouldUpdate = typeof this.statusBarClickBehavior !== 'undefined'
+      this.statusBarClickBehavior = statusBarClickBehavior
+      if (shouldUpdate) {
+        this.update(this.messages)
+      }
+    }))
+    this.subscriptions.add(atom.workspace.observeActivePaneItem(paneItem => {
+      const isTextEditor = atom.workspace.isTextEditor(paneItem)
+      this.element.setVisibility(isTextEditor)
+      if (isTextEditor && this.statusBarRepresents === 'Current File') {
+        this.update(this.messages)
+      }
+    }))
+
+    this.element.onClick((type) => {
+      const workspaceView = atom.views.getView(atom.workspace)
+      if (this.statusBarClickBehavior === 'Toggle Panel') {
+        atom.commands.dispatch(workspaceView, 'linter-ui-default:toggle-panel')
+      } else {
+        atom.commands.dispatch(workspaceView, `linter-ui-default:next-${type}`)
+      }
+    })
+  }
+  update(messages: Array<LinterMessage>): void {
+    this.messages = messages
```

Comment:
📓 It might be worth it to make the parameter optional and just use `this.messages` if it isn't provided to save overwriting `this.messages` with itself... but since the only case where that will happen is when a setting is changed or the initialization it's probably just fine as is.